### ISSUE TITLE
`md` deletes the wrong character on the right-hand side

### ISSUE THREAD

**Issue by Omnikar** [helix-editor/helix#543]
### Reproduction steps
1. Position the cursor within a surround, such as between the parentheses in: `This is (some) text.`
2. Type `md(` or `md)`.
3. The text becomes `This is some)text.`. The left parenthesis was deleted, but the space to the right of the right parenthesis was deleted instead of the parenthesis itself. Likewise, any usage of `md` deletes the character to the right of the character it should delete on the right-hand side.

**Comment by pickfire** [comment]
Yes, this is a known issue on the implementation. I think kakoune might have the same issue, vim probably have a better implementation for this. cc @sudormrfbin

**Comment by sudormrfbin** [comment]
Some of the text objects are broken too, probably a bug after the #376 cleanup. I'll see if I can fix it.

**Comment by cessen** [comment]
@sudormrfbin Sorry to (maybe?) snipe you on this.  I was pretty sure I knew what was wrong.  Should be fixed in #545.

What are the other text objects that are broken?

**Comment by sudormrfbin** [comment]
@cessen Thanks for fixing it :) Around word is the only text object I noticed to be broken; it selects an extra white space on the side it's not supposed to:

```rust
// cursor anywhere on "text" and then `maw`

// expected:
some [text ]here

// current:
some[ text ]here
```

**Comment by cessen** [comment]
@sudormrfbin Okay, cool, I'll fix that too.  Thanks!

**Comment by cessen** [comment]
Fix for word selection: #546

**Comment by pickfire** [comment]
I think our tests should use `[]` or some visual indicator rather than using numbers which is more error prone and we can't easily see stuff.
